src/chips/overflow_check_v2.rs

```diff
@@ -0,0 +1,120 @@
+use std::fmt::Debug;
+
+use super::utils::{
+    range_check_vec,
+};
+use halo2_proofs::{circuit::*, halo2curves::pasta::Fp, plonk::*, poly::Rotation};
+
+#[derive(Debug, Clone)]
+pub struct OverflowCheckV2Config<const MAX_BITS: u8, const ACC_COLS: usize> {
+    pub value: Column<Advice>,
+    pub decomposed_values: [Column<Advice>; ACC_COLS],
+    pub instance: Column<Instance>,
+    pub selector: Selector,
+}
+
+#[derive(Debug, Clone)]
+pub struct OverflowChipV2<const MAX_BITS: u8, const ACC_COLS: usize> {
```

Comment:
I think that `MAX_BITS` doesn't need to be a parameter to instantiate the chip. We can for example impose MAX_BITS to be 4 and the user of chip can decide the range to perform by parametrizing `ACC_COLS`. For example performing a range check of 32 bits will mean using 8 `ACC_COLS` as each column checks for a range of 4 bits.

Reply:
At that time, I created a more flexible configuration for future benchmarking of this chip's performance.
You are right, no need MAX_BITS for now. It is enough for checking overflow by adjusting the column number.